From ef29bfbb9dee59d029a4c8e1e1774b55bd00f74e Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 17 Dec 2008 05:14:00 +0000
Subject: [PATCH 032/259] ENGR00101947 optimize vpu clock gate between
 StartOneFrame and GetOutputInfo

Clock off after decoding/encoding interrupt received immediately,
then clock on again when calling GetOutputInfo.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 6494463..b10fa86 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -998,6 +998,9 @@ RetCode vpu_EncGetOutputInfo(EncHandle handle, EncOutputInfo * info)
 		return RETCODE_INVALID_HANDLE;
 	}
 
+        /* Clock is gated off when received interrupt in driver, so need to gate on here. */
+        IOClkGateSet(true);
+
 	info->picType = VpuReadReg(RET_ENC_PIC_TYPE);
 
 	if (pEncInfo->ringBufferEnable == 0) {
@@ -2253,6 +2256,9 @@ RetCode vpu_DecGetOutputInfo(DecHandle handle, DecOutputInfo * info)
 		vl2cc_flush();
 	}
 
+	/* Clock is gated off when received interrupt in driver, so need to gate on here. */
+	IOClkGateSet(true);
+
 	val = VpuReadReg(RET_DEC_PIC_SUCCESS);
 	info->decodingSuccess = (val & 0x01);
 	if (pCodecInst->codecMode == AVC_DEC) {
-- 
1.8.0

